lib/cocoapods/generator/app_target_helper.rb

```diff
@@ -0,0 +1,226 @@
+      #         the Xcodeproj to generate the target into.
+      #
+      # @param  [PodTarget] pod_target
+      #         the pod target to use for when generating the contents of the import file.
+      #
+      # @param  [Symbol] platform
+      #         the platform of the target. Can be `:ios` or `:osx`, etc.
+      #
+      # @param  [Boolean] use_frameworks
+      #         whether to use frameworks or not when generating the contents of the import file.
+      #
+      # @param  [String] name
+      #         The name of the folder to use and save the generated main file.
+      #
+      # @return [Pathname] the new source file that was generated.
+      #
+      def self.create_app_import_source_file(project, pod_target, platform, use_frameworks, name = 'App')
+        language = pod_target.uses_swift? ? :swift : :objc
+
+        if language == :swift
+          source_file = project.path.dirname.+("#{name}/main.swift")
+          source_file.parent.mkpath
+          import_statement = use_frameworks && pod_target.should_build? ? "import #{pod_target.product_module_name}\n" : ''
+          source_file.open('w') { |f| f << import_statement }
+        else
+          source_file = project.path.dirname.+("#{name}/main.m")
+          source_file.parent.mkpath
+          import_statement = if use_frameworks && pod_target.should_build?
+                               "@import #{pod_target.product_module_name};\n"
+                             else
+                               header_name = "#{pod_target.product_module_name}/#{pod_target.product_module_name}.h"
+                               if pod_target.sandbox.public_headers.root.+(header_name).file?
+                                 "#import <#{header_name}>\n"
+                               else
+                                 ''
+                               end
+                             end
+          source_file.open('w') do |f|
+            f << "@import Foundation;\n"
+            f << "@import UIKit;\n" if platform == :ios || platform == :tvos
+            f << "@import Cocoa;\n" if platform == :osx
+            f << "#{import_statement}int main() {}\n"
+          end
+        end
+        source_file
+      end
+
+      # Creates a default app host 'main.m' file.
+      #
+      # @param  [Project] project
+      #         the Xcodeproj to generate the target into.
+      #
+      # @param  [Symbol] platform
+      #         the platform of the target. Can be `:ios` or `:osx`.
+      #
+      # @param  [String] name
+      #         The name of the folder to use and save the generated main file.
+      #
+      # @return [Pathname] the new source file that was generated.
+      #
+      def self.create_app_host_main_file(project, platform, name = 'App')
+        source_file = project.path.dirname.+("#{name}/main.m")
+        source_file.parent.mkpath
+        source_file.open('w') do |f|
+          case platform
+          when :ios, :tvos
```

Comment:
Works on `:tvos` but generally not for `:watchos` but I think this will be fixed by limiting app hosts at the DSL level. I will open a PR for this after this lands.